pkg/kubelet/kubelet_getters.go

```diff
@@ -140,6 +140,20 @@ func (kl *Kubelet) GetMaxPods() int {
 	return kl.maxPods
 }
 
+func (kl *Kubelet) GetUserNamespacesIDsPerPod() uint32 {
+	userNs := kl.kubeletConfiguration.UserNamespaces
+	if userNs == nil {
+		return config.DefaultKubeletUserNamespacesIDsPerPod
+	}
+	idsPerPod := userNs.IDsPerPod
+	if idsPerPod == nil || *idsPerPod == 0 {
+		return config.DefaultKubeletUserNamespacesIDsPerPod
+	}
+	// The value is already validated to be <= MaxUint32,
+	// so we can safely drop the upper bits.
+	return uint32(*idsPerPod)
```

Comment:
@AkihiroSuda Nothing is blocking it from being a negative value, right? I'm not sure what will happen if this is negative, but I don't want to find out. Can we block it (if we are not doing it already)?

Reply:
@liggitt sorry, I editted the comment while you answer, I guess.

Doing a `uint32(x)` when x is signed, will be a very big number (if x is negative). This is probably unexpected and IMHO we should reject it instead of cast it to uint32. For example: https://go.dev/play/p/Y3T7cf-U_-G

Two's complement makes (signed) negative numbers, when read as unsigned, a a big (unsigned) number, basically.

Reply:
before we get to this point, validation ensures that this value is >=65536 and <MaxUint32

Reply:
Ohhh, I missed that was in the validation, not in the userns_manager (that we have something similar). Perfect, then! Ignore this comment! Sorry!